a11ym

```diff
@@ -250,4 +263,43 @@ crawler.on('fetchcomplete', function(queueItem) {
     urls.push({url: url});
 });
 
-crawler.start();
+var computeURLToCrawler = function (crawler) {
+    var first = true;
+
+    return function (value) {
+        var url = crawler.processURL(value);
+        console.log(url);
+
+        if (true === first) {
+            first                   = false;
+            crawler.host            = url.host;
+            crawler.initialProtocol = url.protocol;
+            crawler.initialPort     = url.port;
+            crawler.initialPath     = url.path;
+        } else {
+            ++maximumUrls;
+        }
+
+        crawler.queue.add(
+            url.protocol,
+            url.host,
+            url.port,
+            url.path
+        );
+    };
+};
+
+if (1 === program.args.length && '-' === program.args[0]) {
```

Comment:
Raising an error? Maybe… or we can just simply ignore it.

If a URL is invalid, it is ignored.

Maybe we can raise a warning if a URL is not valid. I would like to address this in another issue. I am going to create it.